nwplus/livesite
# submit, bug fixes, dashboard messages

### Dashboard message
- updated RSVP-by date to Jan 3rd 2021 (FYI @JPGarCar)
- updated copy text wording and embedded links
- fetching dates from firebase in `InternalWebsites/Livesite`: `sendAcceptancesBy`, `rsvpBy`, `offWaitlistNotify`

### [bugfix] Redirect to `/application/part-1` when application is not submitted
- added condition for `/application` in the `useEffect` of `Auth.js` to check if applicant has submitted their application before being able to access the dashboard - if their application is not submitted, redirect to `/application/part-1` (FYI @andykwan1124)
```
if (location === '/application') {
        await handleUser(setUser, setLocation)
      }
```
- I noticed that the user gets automatically redirected to `/submission` if they RSVP and refresh the page (or navigate away and come back) - I've restricted the redirect to `/submission` for HackCamp only

### [bugfix] User document fields get deleted when you click 'Back' several times in `/application`
- added `!applicationRef.current` in the first check in `forceSave` (thank you @jackyzha0 so big brained 🧠)
```
const forceSave = useCallback(async () => {
    if (!user || !applicationRef.current) return
    const updatedApp = {
      ...applicationRef.current,
      submission: {
        lastUpdated: firebase.firestore.Timestamp.now(),
        submitted: false,
      },
    }
```

### Other minor changes
- changed review page to display `otherEngagementSource` if `Other` is chosen for `engagementSource`
- changed width and border radius of review container to align with design specs
- also fixes bugs listed here https://github.com/nwplus/livesite/pull/220#issuecomment-743953305

### How to test

#### testing dashboard redirect
1. go to https://nwplus-ubc--pr220-date-fix-yjvu589n.web.app/application/part-1 and login
2. go to `/application` - you should be redirected to part 1 of the app
note that after you click on 'submit' on the application, you should no longer get redirected

#### testing dashboard messages 
1. assuming you've logged in, access your Applicant document in the database: https://console.firebase.google.com/u/0/project/nwplus-ubc-dev/firestore/data~2FHackathons~2FnwHacks2021~2FApplicants
2. change your `applicantStatus` to `waitlisted` then go back to `/application` in the preview link https://nwplus-ubc--pr220-date-fix-yjvu589n.web.app/application and refresh - the status and message should update to waitlisted
3. change your `applicantStatus` to `accepted` then go back to preview link and refresh - message should update
